R/Augment_Snapshot.R

```diff
@@ -87,57 +87,58 @@ Augment_Snapshot <- function(
-        snapshot <- read.csv(paste0(snap, "/results_summary.csv")) %>%
-          distinct(.data$workflowid)
-      }, .id = "snapshot_date") %>%
-      group_by(.data$workflowid) %>%
+    status <- stackedSnapshots$results_summary %>%
+      select(.data$snapshot_date, .data$workflowid) %>%
+      filter(.data$snapshot_date %in% snapshot_dates$snapshot_date) %>%
+      distinct() %>%
+      group_by(workflowid) %>%
       summarise(
         latest = max(as.Date(snapshot_date), na.rm = TRUE),
         .groups = "drop"
       ) %>%
-      mutate(is_current = latest == max(latest))
+      mutate(is_current = latest == max(latest)) %>%
+      left_join(snapshot_dates, by = c("latest" = "snapshot_date"))
 
-    old_workflows <- is_current %>%
+    ## get old workflowids
+    old_workflows <- status %>%
       filter(!is_current) %>%
       split(.$latest) %>%
       map(., . %>% pull(.data$workflowid))
 
-    ## pull object into snapshot
-    old_snapshots <- list()
-    for (latest in is_current %>%
-      filter(!.data$is_current) %>%
-      pull(.data$latest) %>%
-      as.character()) {
-      old_snapshots[[latest]] <- readRDS(paste0(nPath, latest, "/snapshot.rds"))
-    }
-
     ## Set active status
     for (kri in names(lSnapshot$lStudyAssessResults)) {
       lSnapshot$lStudyAssessResults[[kri]][["bActive"]] <- TRUE
     }
 
-    for (dates in names(old_snapshots)) {
-      for (kri in names(old_snapshots[[dates]]$lStudyAssessResults)) {
-        old_snapshots[[dates]]$lStudyAssessResults[[kri]][["bActive"]] <- FALSE
+    ## pull object into snapshot
+    if(length(old_workflows) != 0){
+      old_snapshots <- list()
+      for (i in 1:(status %>% filter(!.data$is_current) %>% nrow())) {
+        old_date <- status[i,"latest"] %>% pull() %>% as.character()
+        old_file_path <- paste0(snapshots[grepl(status[i,"foldername"], snapshots)], "/snapshot.rds")
+        old_snapshots[[old_date]] <- readRDS(file = old_file_path)
       }
-    }
 
-    ## Transfer old snaps to current snap
-    for (old_date in names(old_workflows)) {
-      for (kri in old_workflows[[old_date]]) {
-        lSnapshot$lStudyAssessResults[[kri]] <- old_snapshots[[old_date]]$lStudyAssessResults[[kri]]
+      for (dates in names(old_snapshots)) {
+        for (kri in names(old_snapshots[[dates]]$lStudyAssessResults)) {
+          old_snapshots[[dates]]$lStudyAssessResults[[kri]][["bActive"]] <- FALSE
+        }
+      }
+
+      ## Transfer old snaps to current snap
+      for (old_date in names(old_workflows)) {
+        for (kri in old_workflows[[old_date]]) {
+          lSnapshot$lStudyAssessResults[[kri]] <- old_snapshots[[old_date]]$lStudyAssessResults[[kri]]
+        }
       }
     }
-    lSnapshot[["lStatus"]] <- is_current %>%
-      `colnames<-`(c("Workflow ID", "Latest Snapshot", "Currently Active"))
+
+    lSnapshot[["lStatus"]] <- status %>%
+      `colnames<-`(c("Workflow ID", "Latest Snapshot", "Currently Active", "Folder Name"))
```

Comment:
This is nitpicky but I think I'd prefer to use `rename()` here instead of `colnames` just to stay consistent with the tidyverse-style we're using in gsm.